### PR TITLE
More efficient EvmPooledMemory

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Reduce conversions `int` -> `UInt256`
- Single reg256 copy for `SaveWord` rather than invoking `Span.CopyTo`

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs [hunk 1]
```diff
@@ -24,21 +30,34 @@ public class EvmPooledMemory : IEvmMemory
...
+                Unsafe.WriteUnaligned(
+                    ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(_memory), offset),
+                    Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(word))
+                );
+            }
+            else if (word.Length < WordSize)
+            {
+                Array.Clear(_memory!, offset, WordSize - word.Length);
+                word.CopyTo(_memory.AsSpan(offset + WordSize - word.Length, word.Length));
+            }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs:49]: Can this even go this way? I see this comes only from `EvmStack.PopBytes`, maybe that one should return a Vector256 by default and not Span<>? Or we could just ignore this case?

image

File: src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs [hunk 2]
```diff
@@ -24,21 +30,34 @@ public class EvmPooledMemory : IEvmMemory
...
-
-            word.CopyTo(_memory.AsSpan((int)location + WordSize - word.Length, word.Length));
         }
 
         public void SaveByte(in UInt256 location, byte value)
         {
-            CheckMemoryAccessViolation(in location, WordSize);
-            UpdateSize(in location, 1);
+            CheckMemoryAccessViolation(in location, in WordSize256);
+            UpdateSize(in location, in UInt256.One);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs:60]: Minor: both of those calculate `position + length` and are almost always invoked together, maybe we should cover that too?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs]: Done

File: src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs (hunk 2 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs]: Done

**Pull request merged**